ngageoint/opensphere-build-index
# feat(debug): replace debug loader with Closure's

BREAKING CHANGE: The debug loader has been removed in favor of generating
Closure dependencies and using goog.bootstrap. While this requires no changes
to existing index templates, it is a significant change to how applications
load in debug mode so a major version update seems warranted.

**Note:** Requires https://github.com/ngageoint/opensphere-build-closure-helper/pull/9 to test. Once that is merged, I will update the required version in `package.json`.

To test:

- Check out this branch alongside `opensphere` and the updated `opensphere-build-closure-helper`.
- In opensphere, verify `build`, `test`, and `build:debug` scripts all pass.
- Test the debug application to verify files are loaded properly, and ensure the compiled application loads normally (nothing changed there, just a smoke test).
- Build opensphere with plugins that add index files, entry points, etc. Verify those still work as expected.
- Examples of converting existing files to `goog.module` can be found in this branch: https://github.com/schmidtk/opensphere/tree/wip-goog-module. Verify modules load properly in the debug build, can be debugged normally, etc.
